agent/metadata_service.go

```diff
@@ -60,10 +62,17 @@ func makeSecure(handler func(http.ResponseWriter, *http.Request), mds *metadataS
 			return
 		}
 
+		parsedIP := net.ParseIP(ip)
+		allowedIP := ip == `127.0.0.1` || ip == `169.254.169.254`
```

Comment:
With the current construction, I think:
- these defaults should be moved into `NewMetadataService` (as `IP/32` entries)
- any related variable names outside this module should include "extra" or similar indicating that the contents are non-exhaustive (not inside this module as it would be apparent from inspection)